### PR TITLE
Add docs for Client class properties

### PR DESCRIPTION
**Pull request by mille-nium**
Closes: https://github.com/metarhia/impress/issues/1135

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
+//       in application's `www` directory
+//   ext <string> extension of file in the path portion of the request url
+//   typeExt <string> extension type of returned data
+//   data <string> data received from client
+//   context <Object> client context
+//   chunks <Array> used to handle large requests receiving in chunks
+//   ip <string> remote IP address
+//   cookies <Object> received cookies
+//   preparedCookies <string[]> prepared cookies to send
+//   ipInt <number> | <undefined> converted remote IP address to number,
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]:
```suggestion
//   ipInt <number> | <undefined> remote IP address converted to number,
```

File: lib/client.js [hunk 2]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
+//       if no IP address - `undefined`
+//   local <boolean> local network interface flag
+//   session <Session> | <null> client session
+//   sessionCreated <boolean> session creation flag, default: `false`
+//   sessionModified <boolean> session modification flag, default: `false`
+//   logged <boolean> user logged in flag, default: `false`
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]:
```suggestion
//   logged <boolean> whether the user is logged in, default: `false`
```

File: lib/client.js [hunk 3]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
 // HTTP Client interface for Impress Application Server
+// Properties:
+//   startTime <Date> client's start time
+//   req <http.IncomingMessage>
+//   res <http.ServerResponse>
+//   websocket <Object> | <null> initialized
+//       while processing a websocket request
+//   socket <net.Socket> socket associated with the connection
+//   server <Object> | <null> server instance
+//   application <Application> instance of impress.Application
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]:
```suggestion
//   application <Application> instance of <Application>
```

File: lib/client.js [hunk 4]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
+//   dynamicHandler <boolean> flag, true if client has been dispatched,
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]:
```suggestion
//   dynamicHandler <boolean> whether client has been dispatched,
```

File: lib/client.js [hunk 5]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
+//       default: false
+//   query <Object> collection of key and value pairs from url query
+//   schema <string> if server's transport is `tls`,
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]:
```suggestion
//   schema <string> `https` if server's transport is `tls`, otherwise - `http`
```

File: lib/client.js [hunk 6]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
+//       is set to `https`, otherwise - `http`
+//   method <string> the request method in lower case
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]: Nit:
```suggestion
//   method <string> the request method in lowercase
```

File: lib/client.js [hunk 7]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
+//   access <Object> client access options
+//   fields <Object> parsed client data according to request content-type,
+//   files <Object> file uploads object, where the property names
+//       are field names and the values are arrays of file objects
+//   slowTime <number> client slow time in milliseconds
+//   timedOut <boolean> flag, is set to `true` if client hasn't been finished
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]:
```suggestion
//   timedOut <boolean> whether the client hasn't been finished
```

File: lib/client.js [hunk 8]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
+//       in application's `www` directory
+//   execPath <string> path to the nearest directory
+//   execPathDir <string> absolute path to the nearest directory
+//       with handler script file for the request
+//       in application's `www` directory
+//   ext <string> extension of file in the path portion of the request url
+//   typeExt <string> extension type of returned data
+//   data <string> data received from client
+//   context <Object> client context
+//   chunks <Array> used to handle large requests receiving in chunks
```
Comment (SemenchenkoVitaliy) [inline lib/client.js]: IMO this looks more like internal property.

File: lib/client.js [hunk 9]
```diff
@@ -20,6 +20,60 @@ if (!STATUS_CODES[508]) STATUS_CODES[508] = 'Loop Detected';
...
+//   ip <string> remote IP address
```
Comment (belochub) [inline lib/client.js]: Same as the `ipInt` property below, it may be `undefined`.

File: lib/client.js (hunk 8 above)
Comment (mille-nium) [reply lib/client.js]: @SemenchenkoVitaliy so this shouldn't be documented, should it?

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/impress/commit/52cca29548793f8d60e6abb5a4c8addde324efd7

**Pull request closed without merging**